<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>979</width>
    <height>619</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Courier New</family>
    <pointsize>8</pointsize>
    <weight>50</weight>
    <italic>false</italic>
    <bold>false</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>Csound</string>
  </property>
  <property name="styleSheet">
   <string notr="true">font: 8pt &quot;Courier New&quot;;</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="leftMargin">
     <number>4</number>
    </property>
    <property name="topMargin">
     <number>4</number>
    </property>
    <property name="rightMargin">
     <number>4</number>
    </property>
    <property name="bottomMargin">
     <number>4</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QStackedWidget" name="tabs">
       <property name="styleSheet">
        <string notr="true">background-color:lightgray</string>
       </property>
       <widget class="QWidget" name="csdTab">
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QPlainTextEdit" name="csdEdit">
           <property name="font">
            <font>
             <family>Courier New</family>
             <pointsize>8</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);
color: rgb(0, 0, 0);</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="CsoundWebView" name="htmlTab">
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
        </layout>
       </widget>
       <widget class="CsoundWebView" name="manualTab">
        <layout class="QVBoxLayout" name="verticalLayout_3"/>
       </widget>
       <widget class="QWidget" name="portalTab">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QWidget" name="browserToolsWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>24</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>24</height>
            </size>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="spacing">
             <number>2</number>
            </property>
            <property name="leftMargin">
             <number>2</number>
            </property>
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="rightMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QPushButton" name="forwardButton">
              <property name="text">
               <string>Forward</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="backButton">
              <property name="text">
               <string>Back</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="stopLoadingButton">
              <property name="text">
               <string>Stop</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="loadButton">
              <property name="text">
               <string>Load</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string> URL:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="urlEdit">
              <property name="styleSheet">
               <string notr="true">background-color:white;color:black</string>
              </property>
              <property name="text">
               <string>http://csound.github.io/</string>
              </property>
              <property name="placeholderText">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="googleButton">
              <property name="text">
               <string>Google</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="csoundHomeButton">
              <property name="text">
               <string>Csound home</string>
              </property>
             </widget>
            </item>
           </layout>
           <zorder>backButton</zorder>
           <zorder>forwardButton</zorder>
           <zorder>label</zorder>
           <zorder>urlEdit</zorder>
           <zorder>loadButton</zorder>
           <zorder>csoundHomeButton</zorder>
           <zorder>stopLoadingButton</zorder>
           <zorder>googleButton</zorder>
          </widget>
         </item>
         <item>
          <widget class="CsoundWebView" name="portalView" native="true">
           <property name="focusPolicy">
            <enum>Qt::StrongFocus</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="licenseTab">
        <layout class="QVBoxLayout" name="verticalLayout_9">
         <item>
          <widget class="QPlainTextEdit" name="licenseEdit">
           <property name="frameShape">
            <enum>QFrame::Panel</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QPlainTextEdit" name="messagesEdit">
       <property name="styleSheet">
        <string notr="true">background-color: rgb(0, 0, 0);
color: rgb(0, 255, 0);</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAsNeeded</enum>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionSave_as"/>
   <addaction name="actionRun"/>
   <addaction name="actionStop"/>
   <addaction name="actionMakeFullscreen"/>
   <addaction name="separator"/>
   <addaction name="actionShowCsdTab"/>
   <addaction name="actionShowHTMLTab"/>
   <addaction name="actionShowManualTab"/>
   <addaction name="actionShowPortalTab"/>
   <addaction name="actionShowLicenseTab"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>Create new CSD file.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save this CSD file.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as...</string>
   </property>
   <property name="toolTip">
    <string>Save this CSD file with a new filename.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Play</string>
   </property>
   <property name="toolTip">
    <string>Play this CSD file in Csound.</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop running Csound.</string>
   </property>
   <property name="shortcut">
    <string>F6</string>
   </property>
  </action>
  <action name="actionMakeFullscreen">
   <property name="text">
    <string>Fullscreen</string>
   </property>
   <property name="toolTip">
    <string>Make the main window full screen (or not).</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionShowCsdTab">
   <property name="text">
    <string>Editor</string>
   </property>
   <property name="toolTip">
    <string>Show the code editor.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="actionShowHTMLTab">
   <property name="text">
    <string>HTML</string>
   </property>
   <property name="toolTip">
    <string>Show the HTML page.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+2</string>
   </property>
  </action>
  <action name="actionShowManualTab">
   <property name="text">
    <string>Manual</string>
   </property>
   <property name="toolTip">
    <string>Go to the Csound Reference Manual.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+3</string>
   </property>
  </action>
  <action name="actionShowPortalTab">
   <property name="text">
    <string>Portal</string>
   </property>
   <property name="toolTip">
    <string>Go to the Csound home page.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+4</string>
   </property>
  </action>
  <action name="actionShowLicenseTab">
   <property name="text">
    <string>License</string>
   </property>
   <property name="toolTip">
    <string>Show the Csound license.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+5</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CsoundWebView</class>
   <extends>QWidget</extends>
   <header>csoundwebview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>newCsd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openCsd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveCsd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_as</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveCsdAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRun</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>runFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopCsd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowCsdTab</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showCsdTab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowHTMLTab</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showHtmlTab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowManualTab</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showManualTab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowPortalTab</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showPortalTab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowLicenseTab</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showLicenseTab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionMakeFullscreen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>makeFullScreen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateMessages(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_updateMessages(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>489</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>updateMessages(QString)</signal>
  <slot>newCsd()</slot>
  <slot>openCsd()</slot>
  <slot>saveCsd()</slot>
  <slot>saveCsdAs()</slot>
  <slot>runCsd()</slot>
  <slot>stopCsd()</slot>
  <slot>makeFullScreen()</slot>
  <slot>showCsdTab()</slot>
  <slot>showHtmlTab()</slot>
  <slot>showManualTab()</slot>
  <slot>showPortalTab()</slot>
  <slot>showLicenseTab()</slot>
  <slot>on_updateMessages(QString)</slot>
 </slots>
</ui>
